.github/workflows/main.yml

```diff
@@ -22,102 +22,100 @@ on:
-        fetch-depth: 0 # all
-
-    - name: Setup NET 8.0
-      uses: actions/setup-dotnet@v3
-      with:
-        dotnet-version: '8.0.x'
-
-    # Adjustment is done prior to Nuke build as OCTOVERSION information is included in the result package.
-    - name: Append OCTOVERSION_CurrentBranch with -nightly (for scheduled)
-      if: github.event_name == 'schedule'
-      run: echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly-$(date +'%Y%m%d%H%M%S')" >> $env:GITHUB_ENV
-
-    - name: Nuke Build 🏗
-      id: build
-      run: ./build.ps1 --verbosity verbose
-
-    - name: Windows unit test report
-      uses: dorny/test-reporter@v1
-      if: success() || failure()    # run this step even if previous step failed
-      with:
-        name: Windows unit test results
-        path: ./TestResults/*.trx
-        reporter: dotnet-trx
-        fail-on-error: true
-
-    - name: Tag release (when not pre-release) 🏷️
-      if: ${{ !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
-      uses: actions/github-script@v7
-      with:
-        github-token: ${{ github.token }}
-        script: |
-          github.rest.git.createRef({
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            ref: "refs/tags/${{ steps.build.outputs.octoversion_fullsemver }}",
-            sha: context.sha
-          })
-
-    - name: Login to Octopus Deploy 🐙
-      if: (! contains(github.ref, '/merge')) && (! contains(github.ref, '/dependabot/')) && (! contains(github.ref, 'prettybot/'))
-      uses: OctopusDeploy/login@v1
-      with: 
-        server: https://deploy.octopus.app
-        service_account_id: 44daf805-da46-4efb-a403-a3b656dc31fc
-
-    - name: Push to Octopus 🐙
-      uses: OctopusDeploy/push-package-action@v3
-      if: (! contains(github.ref, '/merge')) && (! contains(github.ref, '/dependabot/')) && (! contains(github.ref, 'prettybot/'))
-      with:
-        packages: |
-          ./artifacts/Octopus.Octodiff.${{ steps.build.outputs.octoversion_fullsemver }}.nupkg
-
-    - name: Create Release in Octopus 🐙
-      uses: OctopusDeploy/create-release-action@v3
-      if: (! contains(github.ref, '/merge')) && (! contains(github.ref, '/dependabot/')) && (! contains(github.ref, 'prettybot/'))
-      with:
-        project: Octodiff
-        packages: |
-          Octopus.Octodiff:${{ steps.build.outputs.octoversion_fullsemver }}
+      # Must clone the entire history for OctoVersion to work
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 
+
+      - name: Setup .NET 8
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+
+      - name: Append OCTOVERSION_CurrentBranch with -nightly-<timestamp> (for scheduled)
+        if: github.event_name == 'schedule'
+        run: echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly-$(Get-Date -Format 'yyyyMMddHHmmss')" >> $env:GITHUB_ENV
```

Comment:
Previously we copy-pasted `$(date +'%Y%m%d%H%M%S')` which is valid bash, but doesn't work at all in PowerShell when I run it locally as it isn't valid Powershell.
However, running on GHA it somehow worked under powershell ?!?!? Maybe they parse the `env` file through bash or something?

https://github.com/OctopusDeploy/Octodiff/actions/runs/7804730098/job/21287487639

This corrects it to use powershell syntax anyway

Reply:
I think they just have GNU coreutils installed on the Windows agents. For things like `echo` that has no effect because `echo` is an actual alias:

```powershell
PS > Get-Alias echo

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Alias           echo -> Write-Output                                          
```

but `date`, despite executing `Get-Date` by default on Windows, isn't actually an alias:

```powershell
PS > Get-Alias date  
Get-Alias: This command cannot find a matching alias because an alias with the name 'date' does not exist.
```

If a command doesn't exist Powershell will try `Get-<command>` before giving up which is why it runs `Get-Date` for you, resulting in an invalid syntax error. Aliases take first preference but since `date` isn't an alias, if you have a `date` binary in `$PATH`, it will prefer to use that over the `Get-<command>` fallback. So if GNU utils are installed, that would probably explain this.